kolibri/core/notifications/api.py

```diff
@@ -286,32 +285,50 @@ def parse_attemptslog(attemptlog):
     for the user & resource
     """
     # This Event should not be triggered when a Learner is interacting with an Exercise outside of a Lesson:
-    lessons = get_assignments(attemptlog, attemptlog.masterylog.summarylog, attempt=True)
+    lessons = get_assignments(attemptlog.user, attemptlog.masterylog.summarylog, attempt=True)
     if not lessons:
         return
     # get all the attempt logs on this exercise:
     failed_interactions = []
     attempts = AttemptLog.objects.filter(masterylog_id=attemptlog.masterylog_id)
+
+    # NOTE: saw at elast one error here where failed['correct'] raised a key error
     failed_interactions = [failed for attempt in attempts for failed in attempt.interaction_history if failed['correct'] == 0]
+
     # More than 3 errors in this mastery log:
     needs_help = len(failed_interactions) > 3
 
-    if needs_help:
-        notifications = []
-        for lesson, contentnode_id in lessons:
+    notifications = []
+    for lesson, contentnode_id in lessons:
+        if needs_help:
```

Comment:
why have the needs_help moved to check every lesson? I can't think  how needs_help can affect only to one lesson if the needs come by the attemptlog.

Reply:
Because I am also now doing the 'started' notification checking for exercises only during the attempt log parsing too. So, the behaviour is the same for the needs help behaviour (the tests still pass!), it's just that we use the same loop to also check the started notifications too.